pyracf/common/security_admin.py

```diff
@@ -108,61 +132,40 @@ def _extract_and_check_result(
     def _make_request(
         self,
         security_request: SecurityRequest,
-        irrsmo00_options: int = 1,
-    ) -> Union[dict, bytes]:
-        """Make request to IRRSMO00."""
-        redact_password = (
-            self.__preserved_segment_traits.get("base", {})
-            .get("base:password", {})
-            .get("value", "")
-        )
-        redact_passphrase = (
-            self.__preserved_segment_traits.get("base", {})
-            .get("base:passphrase", {})
-            .get("value", "")
-        )
-        return self.__make_request_and_redact_secrets(
-            security_request,
-            irrsmo00_options=irrsmo00_options,
-            redact_strings=[redact_password, redact_passphrase],
-        )
-
-    def __make_request_and_redact_secrets(
-        self,
-        security_request: SecurityRequest,
-        irrsmo00_options: int = 1,
-        redact_strings: List[str] = [],
+        irrsmo00_precheck: bool = False,
     ) -> Union[dict, bytes]:
         """
         Make request to IRRSMO00.
-        Note: Passwords are redacted from result and from log messages.
+        Note: Secrets are redacted from all data returned to the user and log messages.
         """
         if self.__debug:
             self.__logger.log_dictionary(
                 "Request Dictionary",
                 self.__preserved_segment_traits,
-                redact_strings=redact_strings,
+                secret_traits=self.__secret_traits,
             )
             self.__logger.log_xml(
                 "Request XML",
                 security_request.dump_request_xml(encoding="utf-8"),
-                redact_strings=redact_strings,
+                secret_traits=self.__secret_traits,
             )
         if self.__generate_requests_only:
-            return self.__logger.redact_strings(
+            request = self.__logger.redact_request_xml(
                 security_request.dump_request_xml(encoding="utf-8"),
-                redact_strings=redact_strings,
+                secret_traits=self.__secret_traits,
             )
-        result_xml = self.__irrsmo00.call_racf(
-            security_request.dump_request_xml(), irrsmo00_options
+            self.__clear_state(security_request)
+            return request
+        result_xml = self.__logger.redact_result_xml(
+            self.__irrsmo00.call_racf(
+                security_request.dump_request_xml(), irrsmo00_precheck
+            ),
+            self.__secret_traits,
         )
+        self.__clear_state(security_request)
         if self.__debug:
-            self.__logger.log_xml(
-                "Result XML", result_xml, redact_strings=redact_strings
-            )
-        results = SecurityResult(
-            self.__logger.redact_strings(result_xml, redact_strings=redact_strings)
-        )
+            self.__logger.log_xml("Result XML", result_xml)
```

Comment:
Might be good to have a comment here that briefly explains why this doesn't need to be redacted.

Reply:
sure, added the following:

```python
            # No need to redact anything here since the raw result dictionary
            # already has secrets redacted when passed to logger
            self.__logger.log_xml("Result XML", result_xml)
```

Reply:
agree, that's a less confusing way to explain this. change made